### PR TITLE
Generate entities for new API

### PR DESCRIPTION
**Pull request by helto4real**
## Proposed change

This PR fixes Entity generation for new Rx API.

This works a bit differently from original API. It creates a base class derived from NetDaemonRxApp that you should use in your apps instead. This makes it more convenient to use your generated entities.

Example:
```c#
Light.Livingroom.TurnOn();
MediaPlayer.Kitchen("play_media");
```

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [X] The code change is tested and works locally.
- [X] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [ x I have followed the [development checklist][dev-checklist]
- [x] The code compiles without warnings (code quality chek)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/net-daemon/docs

### CODE REVIEW

File: src/DaemonRunner/DaemonRunner/Service/App/CodeGenerator.cs [hunk 1]
```diff
@@ -100,6 +100,77 @@ public class CodeGenerator
             return code.NormalizeWhitespace(indentation: "    ", eol: "\n").ToFullString();
         }
 
+        public string? GenerateCodeRx(string nameSpace, IEnumerable<string> entities)
+        {
+            var code = SyntaxFactory.CompilationUnit();
+
+            // Add Usings statements
+            code = code.AddUsings(SyntaxFactory.UsingDirective(SyntaxFactory.ParseName("JoySoftware.HomeAssistant.NetDaemon.Common.Reactive")));
```
Comment (BeeHiveJava) [inline src/DaemonRunner/DaemonRunner/Service/App/CodeGenerator.cs:108]: If we can use a ```typeof(SOME_CLASS_IN_REACTIVE_NAMESPACE).Namespace``` here, it'll make refactoring easier.

File: src/DaemonRunner/DaemonRunner/Service/App/CodeGenerator.cs [hunk 2]
```diff
@@ -100,6 +100,77 @@ public class CodeGenerator
...
+
+            // Add namespace
+            var namespaceDeclaration = SyntaxFactory.NamespaceDeclaration(SyntaxFactory.ParseName(nameSpace)).NormalizeWhitespace();
+
+            // Add support for extensions for entities
+            var extensionClass = SyntaxFactory.ClassDeclaration("GeneratedAppBase");
```
Comment (BeeHiveJava) [inline src/DaemonRunner/DaemonRunner/Service/App/CodeGenerator.cs:114]: ```nameof```?

File: src/DaemonRunner/DaemonRunner/Service/App/CodeGenerator.cs [hunk 3]
```diff
@@ -100,6 +100,77 @@ public class CodeGenerator
...
+
+            extensionClass = extensionClass.AddModifiers(SyntaxFactory.Token(SyntaxKind.PublicKeyword));
+            extensionClass = extensionClass.AddBaseListTypes(
+                SyntaxFactory.SimpleBaseType(SyntaxFactory.ParseTypeName("NetDaemonRxApp")));
```
Comment (BeeHiveJava) [inline src/DaemonRunner/DaemonRunner/Service/App/CodeGenerator.cs:118]: ```nameof```?

File: src/DaemonRunner/DaemonRunner/Service/App/CodeGenerator.cs [hunk 4]
```diff
@@ -100,6 +100,77 @@ public class CodeGenerator
...
+            namespaceDeclaration = namespaceDeclaration.AddMembers(extensionClass);
+
+            // Add the classes implementing the specific entities
+            foreach (var domain in GetDomainsFromEntities(entities))
+            {
+                if (_FluentApiMapper.ContainsKey(domain))
+                {
+                    var classDeclaration = $@"public partial class {domain.ToCamelCase()}Entities
+    {{
+        private readonly NetDaemonRxApp _app;
```
Comment (BeeHiveJava) [inline src/DaemonRunner/DaemonRunner/Service/App/CodeGenerator.cs:142]: ```nameof``` can be used here as well I think.

File: src/DaemonRunner/DaemonRunner/Service/App/CodeGenerator.cs (hunk 1 above)
Comment (helto4real) [reply src/DaemonRunner/DaemonRunner/Service/App/CodeGenerator.cs:108]: Sorry I missed your comments before merge.. Preparing a new PR to fix them now

File: src/DaemonRunner/DaemonRunner/Service/App/CodeGenerator.cs (hunk 2 above)
Comment (helto4real) [reply src/DaemonRunner/DaemonRunner/Service/App/CodeGenerator.cs:114]: It does not exist so I can not use that here.. It is a class that are generated

**Pull request merged**